src/library/scala/collection/immutable/Vector.scala

```diff
@@ -113,22 +106,18 @@ override def companion: GenericCompanion[Vector] = Vector
       } else Iterator.empty.next()
   }
 
-  // TODO: reverse
-
-  // TODO: check performance of foreach/map etc. should override or not?
   // Ideally, clients will inline calls to map all the way down, including the iterator/builder methods.
   // In principle, escape analysis could even remove the iterator/builder allocations and do it
   // with local variables exclusively. But we're not quite there yet ...
 
   def apply(index: Int): A = {
     val idx = checkRangeConvert(index)
-    //println("get elem: "+index + "/"+idx + "(focus:" +focus+" xor:"+(idx^focus)+" depth:"+depth+")")
     getElem(idx, idx ^ focus)
   }
 
   private def checkRangeConvert(index: Int) = {
     val idx = index + startIndex
-    if (0 <= index && idx < endIndex)
+    if (index >= 0 && idx < endIndex)
```

Comment:
@Ichoran, this makes sure all indices are strictly non-negative.
(switched the comparison order to make it more intuitive)